### PR TITLE
stdio: Implement rename

### PR DESCRIPTION
**Pull request by dlrobertson**

### CODE REVIEW

File: src/platform/src/redox/mod.rs [hunk 1]
```diff
@@ -184,6 +184,17 @@ pub fn read(fd: c_int, buf: &mut [u8]) -> ssize_t {
     e(syscall::read(fd as usize, buf)) as ssize_t
 }
 
+pub fn rename(old: *const c_char, new: *const c_char) -> c_int {
+    let (old_path, new_path) = unsafe { (c_str(old), c_str(new)) };
+    match syscall::open(old_path, O_WRONLY) {
+        Ok(fd) => match syscall::frename(fd, new_path) {
```
Comment (jackpot51) [inline src/platform/src/redox/mod.rs]: You need to close the fd if frename fails

Comment (dlrobertson) [reply src/platform/src/redox/mod.rs]: :+1: good catch

Comment (dlrobertson) [reply src/platform/src/redox/mod.rs]: fixed

Comment (jackpot51) [reply src/platform/src/redox/mod.rs]: Also, the return value from frename should be returned, not the one from close

Comment (jackpot51) [reply src/platform/src/redox/mod.rs]: It would be better to do:

let res = syscall::frename(fd, new_path);
let _ = syscall::close(fd);
e(res)

Comment (dlrobertson) [reply src/platform/src/redox/mod.rs]: :+1: updated

**Pull request merged**